packages/angular/cli/src/commands/generate/cli.ts

```diff
@@ -162,4 +159,54 @@ export class GenerateCommandModule
 
     return `${commandName}${positionalArgs ? ' ' + positionalArgs : ''}`;
   }
+
+  /**
+   * Get schematics that needs to be registered as subcommands.
+   *
+   * @returns a sorted list of schematics that needs to be registered as subcommands.
+   */
+  private async getSchematicsToRegister(): Promise<
+    [schematicName: string, collectionName: string][]
+  > {
+    const schematicsToRegister = new Map<string, string>();
+    const collectionNames = await this.getCollectionNames();
+    const [, schematicNameFromArgs] = this.parseSchematicInfo(
+      // positional = [generate, component] or [generate]
+      this.context.args.positional[1],
+    );
+
+    for (const collectionName of collectionNames) {
+      const workflow = this.getOrCreateWorkflowForBuilder(collectionName);
+      const collection = workflow.engine.createCollection(collectionName);
+
+      for (const schematicName of collection.listSchematicNames(true /** includeHidden */)) {
+        if (schematicName === schematicNameFromArgs) {
+          // We located the schematic the user invoked.
+          return [[schematicName, collectionName]];
+        }
+
+        // If a schematic with this same name is already registered skip.
+        if (!schematicsToRegister.has(schematicName)) {
+          schematicsToRegister.set(schematicName, collectionName);
+        }
+      }
+    }
+
+    // Didn't find the schematic or no schematic name was provided Ex: `ng generate --help`.
+    return Array.from(schematicsToRegister).sort((a, b) => {
+      // ignore casing
+      const nameA = a[0].toUpperCase();
+      const nameB = b[0].toUpperCase();
```

Comment:
Nit: Could this be written as a destructure? That way it's a little more self-descriptive about what values are being compared.

```typescript
Array.from(schematicsToRegister).sort(([key1], [key2]) => /* ... */);
```

Reply:
Done